pkg/dcgmexporter/exporter_metrics.go

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dcgmexporter
+
+import "fmt"
+
+const (
+	dcgmExpClockThrottleReasonsCount = "DCGM_FI_EXP_CLOCK_THROTTLE_REASONS_COUNT"
+	dcgmExpXIDErrorsCount            = "DCGM_EXP_XID_ERRORS_COUNT"
+)
+
+type ExporterCounter uint16
+
+const (
+	DCGMFIUnknown                 ExporterCounter = 0
+	DCGMXIDErrorsCount            ExporterCounter = iota + 9000
+	DCGMClockThrottleReasonsCount ExporterCounter = iota
+)
+
+// String method to convert the enum value to a string
```

Comment:
Similar to the earlier comments, the standard practice is to have consts and variables in `config.go` and new structs/types in `types.go`

It can accompany comment below similar to below:
```
/**************************************
* ExporterCounter comment here
***************************************/
type ExporterCounter uint16

// String method to convert the enum value to a string
func (enm ExporterCounter) String() string {
	switch enm {
	case DCGMXIDErrorsCount:
		return dcgmExpXIDErrorsCount
	case DCGMClockThrottleReasonsCount:
		return dcgmExpClockThrottleReasonsCount
	default:
		return "DCGM_FI_UNKNOWN"
	}
}

func IdentifyMetricType(s string) (ExporterCounter, error) {
	mv, ok := DCGMFields[s]
	if !ok {
		return mv, fmt.Errorf("Unknown ExporterCounter field '%s'", s)
	}
	return mv, nil
}
```

Reply:
Domain Driven Design (DDD) has another practice where we describe a domain model and use Ubiqutios language in the code, files, etc. I strongly believe we need to follow the DDD approach because it is much easier to explain, communicate, and maintain.